pip/commands/configuration.py

```diff
@@ -0,0 +1,242 @@
+            dest='edit',
+            action='store_true',
+            default=False,
+            help='Edit the configuration file'
+        )
+
+        self.cmd_opts.add_option(
+            '--get',
+            dest='get_name',
+            action='store',
+            metavar='name',
+            default=None,
+            help='Get the value associated with name in the configuration file'
+        )
+
+        self.cmd_opts.add_option(
+            '--set',
+            dest='set_name_value',
+            action='store',
+            metavar='name=value',
+            type="string",  # this is validated elsewhere
+            default=None,
+            help='Set name=value in the configuration file'
+        )
+
+        self.cmd_opts.add_option(
+            '--unset',
+            dest='unset_name',
+            action='store',
+            metavar='name',
+            default=None,
+            help=(
+                'Unset the value associated with name in the configuration '
+                'file'
+            )
+        )
+
+        self.cmd_opts.add_option(
+            '--global',
+            dest='global_file',
+            action='store_true',
+            default=False,
+            help='Use the system-wide configuration file only'
+        )
+
+        self.cmd_opts.add_option(
+            '--user',
+            dest='user_file',
+            action='store_true',
+            default=False,
+            help='Use the user configuration file only'
+        )
+
+        self.cmd_opts.add_option(
+            '--venv',
+            dest='venv_file',
+            action='store_true',
+            default=False,
+            help='Use the virtualenv configuration file only'
+        )
+
+        self.parser.insert_option_group(0, self.cmd_opts)
+
+    def run(self, options, args):
+
+        # Determine what action is to be taken
+        action_options = {
+            "list": options.list,
+            "edit": options.edit,
+            "get": options.get_name,
+            "set": options.set_name_value,
+            "unset": options.unset_name,
+        }
+
+        action = None
+        for k, v in action_options.items():
+            if v:
+                if action is not None:
+                    # this works because of the conditional immediately after
+                    # the loop.
+                    action = None
+                    break
+                action = k
+
+        if action is None:
+            logger.error(
+                "Need exactly one action (--list, --edit, "
+                "--get, --set, --unset) to perform."
+            )
+            self.parser.print_help()
+            return ERROR
+
+        # Determine which configuration files are to be loaded
+        if sum([options.user_file, options.global_file, options.venv_file]):
+            logger.error(
+                "Need at-most one configuration file to use - pass "
+                "only one of --global, --user, --venv."
```

Comment:
The only time assuming a file could be useful is when we have to modify something. At the other times - list and get, it makes more sense to be asking the user to specify if they want to load a specific file - in which case we only load that file.

I feel it makes more sense to require the user to pass which file is to be modified or used since it gives them the ability to inspect the effect of each level or overridding as well as a fine degree of control.